src/session/sessions.ts

```diff
@@ -0,0 +1,249 @@
+   */
+  async send(token: string, data: DocumentNode, variables?: object) {
+    let headers = {
+      authorization: token,
+    };
+
+    return this.ep.send("query", data, variables, headers);
+  }
+}
+
+/** @class A Snek SubSession  */
+class SnekSession extends Session {
+  refreshToken: string | undefined = "";
+  refreshTokenName: string = "refresh";
+
+  /**
+   * Define tasks
+   */
+  public tasks: SnekTasks;
+
+  /**
+   * Creates an instance of a SnekSession.
+   *
+   * @constructor
+   * @param {string} sId A session name
+   * @param {Endpoint} ep A endpoint
+   * @param {SnekTemplate} template A template set
+   */
+  constructor(
+    sId: string,
+    public ep: ApolloEndpoint,
+    public template: SnekTemplate
+  ) {
+    super(sId);
+
+    this.tokenName = sId + "-" + this.tokenName;
+    this.refreshTokenName = sId + "-" + this.refreshTokenName;
+    this.tasks = new SnekTasks(this);
+  }
+
+  //> Methods
+  /**
+   * Send query:
+   *
+   * @description Send a query to the endpoint.
+   * @param {string} token A authentication token.
+   * @param {DocumentNode} data A DocumentNode with a query.
+   * @param {object} variables A abject with variables.
+   */
+  async send(token: string, data: DocumentNode, variables?: object) {
+    let headers = {
+      authorization: token,
+    };
+
+    return this.ep.send("query", data, variables, headers);
+  }
+
+  /**
+   * Initialize tokens.
+   *
+   * @param {IAuth} auth A Auth object (token, refreshToken).
+   */
+  initTokens(auth: IAuth) {
+    this.token = auth.token;
+    this.refreshToken = auth.refreshToken;
+
+    /** Delte the token and refreshToken cookie. */
+    deleteCookie(this.tokenName);
+    deleteCookie(this.refreshTokenName);
+    /** Set the token and refreshToken cookie with expire times. */
+    setCookie(this.tokenName, this.token, 2 * 60);
+    setCookie(this.refreshTokenName, this.refreshToken, 6 * 24 * 60 * 60);
+  }
+
+  /**
+   * Was alive check.
+   *
+   * @description Refresh token status check.
+   * @param {boolean} alive A status whether the refresh token is alive or not
+   */
+  wasAlive() {
+    return cookieChecker(this.refreshTokenName);
+  }
+
+  /**
+   * Is alive check.
+   *
+   * @description Token and refresh token status check.
+   * @param {boolean} alive A status whether the token and refresh token is alive or not
+   */
+  isAlive() {
+    if (cookieChecker(this.refreshTokenName) && super.isAlive()) {
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Begin session.
+   *
+   * @param {string} user A User defined by username and password.
+   * @returns {UserData} A UserData object.
+   */
+  async begin(user?: User) {
+    let response;
+
+    if (!user && this.wasAlive()) {
+      /** Refresh tokens. */
+      this.refresh();
+    } else {
+      if (!user) {
+        /** Authenticate anonymous user. */
+        response = await this.tasks.auth.anon();
+      } else {
+        /** Authenticate real user. */
```

Comment:
Please consider using our comment style for single line comments.
`// Comment`
Single line comments shall not have a period if it contains less than two sentences.